server/app/auth/DefaultToGuestRedirector.java

```diff
@@ -0,0 +1,19 @@
+package auth;
+
+import static controllers.CallbackController.REDIRECT_TO_SESSION_KEY;
+import static play.mvc.Results.redirect;
+
+import com.google.common.collect.ImmutableMap;
+import controllers.routes;
+import play.mvc.Http;
+import play.mvc.Result;
+
+public final class DefaultToGuestRedirector {
+
+  public static Result createGuestSessionAndRedirect(Http.Request request) {
+    // If there isn't currently a session, create a guest session using the CallbackController,
+    // and add a session key that asks it to redirect back here when the profile will be present.
+    return redirect(routes.CallbackController.callback(GuestClient.CLIENT_NAME).url())
+        .withSession(ImmutableMap.of(REDIRECT_TO_SESSION_KEY, request.uri()));
```

Comment:
The `.withSession(ImmutableMap.of(REDIRECT_TO_SESSION_KEY, request.uri()))` line wasn't there for `HomeController` before since it wasn't required, but there is no harm in adding it for consistency.